rusk/src/lib/http/driver.rs

```diff
@@ -0,0 +1,283 @@
+        init.call(store, ())?;
+        Ok(())
+    }
+
+    pub fn contract_id(&self) -> ContractId {
+        self.contract_id
+    }
+
+    fn allocate(
+        &self,
+        store: &mut Store<()>,
+        sz: usize,
+    ) -> Result<*mut u8, Error> {
+        let alloc = self
+            .instance
+            .get_typed_func::<i32, i32>(&mut *store, "alloc")
+            .map_err(|e| Error::Other(format!("allocate failed: {e}")))?;
+        let mem = alloc
+            .call(&mut *store, sz as i32)
+            .map_err(|e| Error::Other(format!("allocate failed: {e}")))?;
+        Ok(mem as *mut u8)
+    }
+
+    fn allocate_and_copy(
+        &self,
+        store: &mut Store<()>,
+        bytes: &[u8],
+        sz: usize,
+    ) -> Result<*mut u8, Error> {
+        let mem = self.allocate(&mut *store, sz)?;
+        let wasm_memory = self
+            .instance
+            .get_memory(&mut *store, "memory")
+            .ok_or(Error::Other("getting memory failed".to_string()))?;
+        wasm_memory
+            .write(&mut *store, mem as usize, bytes)
+            .map_err(|e| Error::Other(format!("allocate failed: {e}")))?;
+        Ok(mem)
+    }
+
+    fn deallocate(
+        &self,
+        store: &mut Store<()>,
+        ptr: *mut u8,
+        sz: usize,
+    ) -> Result<(), Error> {
+        let dealloc = self
+            .instance
+            .get_typed_func::<(i32, i32), ()>(&mut *store, "dealloc")
+            .map_err(|e| Error::Other(format!("deallocate failed: {e}")))?;
+        dealloc
+            .call(&mut *store, (ptr as i32, sz as i32))
+            .map_err(|e| Error::Other(format!("deallocate failed: {e}")))?;
+        Ok(())
+    }
+
+    // Reads bytes from a given memory pointer assuming that
+    // first 4 bytes hold little endian-encoded buffer length L.
+    // Subsequent bytes form a buffer of length L
+    //
+    // Returns: vector containing copy of the buffer
+    fn read_u32_le_and_bytes(
+        &self,
+        store: &mut Store<()>,
+        p: *const u8,
+    ) -> Result<Vec<u8>, Error> {
+        let wasm_memory = self
+            .instance
+            .get_memory(&mut *store, "memory")
+            .ok_or(Error::Other("getting memory failed".to_string()))?;
+
+        let mut buf_len_buf = [0u8; 4];
+        wasm_memory
+            .read(&mut *store, p as usize, &mut buf_len_buf)
+            .map_err(|e| {
+                Error::Other(format!("reading wasm memory failed: {e}"))
+            })?;
+        let buf_len = u32::from_le_bytes(buf_len_buf);
+
+        let data_ptr = unsafe { p.add(4) };
+
+        let mut buffer = vec![0u8; buf_len as usize];
+        wasm_memory
+            .read(&mut *store, data_ptr as usize, &mut buffer)
+            .map_err(|e| {
+                Error::Other(format!("reading wasm memory failed: {e}"))
+            })?;
+
+        Ok(buffer)
+    }
+
+    fn decoding_fn(
+        &self,
+        fn_name: &str,
```

Comment:
`fn_name` doesn't quite fit in the case of event decoding... maybe it would better to use a more generic name, like `data_name` or `object_name`